src/proxy.rs

```diff
@@ -165,28 +167,18 @@ impl ProxyHandlerWebSocket {
         // Stream frontend messages to backend.
         let stream_to_backend = async move {
             while let Some(Ok(msg_axm)) = frontend_stream.next().await {
-                // Ugh this is a lot of work that could be avoided if
-                // there were some way to get the wrapped message out
-                // directly, but there isn't...
-                let msg_tng = if msg_axm.is_binary() {
-                    MsgTng::Binary(msg_axm.as_bytes().to_owned())
-                } else if let Some(str) = msg_axm.to_str() {
-                    MsgTng::Text(str.to_owned())
-                } else if msg_axm.is_ping() {
-                    MsgTng::Ping(msg_axm.as_bytes().to_owned())
-                } else if msg_axm.is_pong() {
-                    MsgTng::Pong(msg_axm.as_bytes().to_owned())
-                } else if let Some((code, reason)) = msg_axm.close_frame() {
-                    MsgTng::Close(Some(CloseFrame {
-                        code: code.into(),
-                        reason: Cow::Owned(reason.to_owned()),
-                    }))
-                } else if msg_axm.is_close() {
-                    MsgTng::Close(None)
-                } else {
-                    tracing::error!(msg = ?msg_axm, "Received message is not binary, text, ping, pong, or close");
-                    return;
+                let msg_tng = match msg_axm {
+                    MsgAxm::Text(msg) => MsgTng::Text(msg.to_owned()),
+                    MsgAxm::Binary(msg) => MsgTng::Binary(msg),
+                    MsgAxm::Ping(msg) => MsgTng::Ping(msg),
+                    MsgAxm::Pong(msg) => MsgTng::Pong(msg),
+                    MsgAxm::Close(Some(close_frame)) => MsgTng::Close(Some(CloseFrame {
+                        code: close_frame.code.into(),
+                        reason: close_frame.reason,
+                    })),
+                    MsgAxm::Close(None) => MsgTng::Close(None),
```

Comment:
`axum::extract::ws::Msg` is now an enum :)

Reply:
Love it!